src/agent/onefuzz-result/src/job_result.rs

```diff
@@ -0,0 +1,152 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use anyhow::Result;
+use async_trait::async_trait;
+use reqwest::Url;
+use serde::{self, Deserialize, Serialize};
+use std::collections::HashMap;
+use std::{
+    collections::HashSet,
+    sync::{Arc, Mutex},
+};
+use storage_queue::QueueClient;
+use tokio::sync::Notify;
+use uuid::Uuid;
+
+#[derive(Debug, Deserialize, Serialize, Hash, Eq, PartialEq, Clone)]
+#[serde(tag = "type")]
+pub enum JobResultData {
+    NewCrashingInput,
+    NoReproCrashingInput,
+    NewReport,
+    NewUniqueReport,
+    NewRegressionReport,
+    NewCoverage,
+    CoverageData,
+    RuntimeStats,
+}
+
+#[derive(Debug, Deserialize, Serialize, Clone)]
+struct JobResult {
+    task_id: Uuid,
+    job_id: Uuid,
+    machine_id: Uuid,
+    machine_name: String,
+    data: Vec<JobResultData>,
+    value: HashMap<String, f64>,
+}
+
+#[derive(Clone)]
+pub struct TaskContext {
+    task_id: Uuid,
+    job_id: Uuid,
+    machine_id: Uuid,
+    machine_name: String,
+}
+
+pub struct JobResultContext<TContext, T> {
+    pub state: TContext,
+    pub queue_client: QueueClient,
+    pub pending_messages: Mutex<HashSet<T>>,
```

Comment:
Do we ever use this? It looks like we directly queue the messages and never put them in this hashset.

Reply:
Nope, another leftover. Removing.